Chromatic 3/8/22 build

hnguyen1179 opened the pull request.

dependabot[bot] committed:
Bump eslint-config-prettier from 8.4.0 to 8.5.0 (#538)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

hnguyen1179 committed:
Add ELFI icon and shorten decimal places  (#539)

* Change 4 decimal to 1 decimal place

* Add ELFI icon & revise html structure

* Add ELFI icon to row

* Fix text stacking issue

* Fix pr

hnguyen1179 committed:
Switch button positions (#540)

hnguyen1179 committed:
Add snapshot testing via chromatic & accessibility check in sb (#528)

* Add Chromatic

* Deoptimize next.js image for usage in sb

* Add snapshot testing & ui pr review

* Add accessibility check to sb

* Add newline

* Fix typo

* Update package.json

* Sync package-lock.json

* Remove build-storybook.log from tracking

* Fix merge issues

* Fix icon width

* Add documentation

* Enable chromatic CI to run on chromatic-tests branch

hnguyen1179 committed:
Remove GSC icons (#541)

hnguyen1179 committed:
Remove auto-scroll feature (#542)

* Remove auto-scrolling feature

* Delete unused code

hnguyen1179 committed:
Remove logic for success/fail message within address input (#543)

* Remove logic for success/fail message on input

* Fix build error

dependabot[bot] committed:
Bump @tailwindcss/forms from 0.4.0 to 0.5.0 (#544)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

hnguyen1179 committed:
Link to etherscan in toast notifications (#545)

* Touch up docs

* Add etherscan link to tx loading toast

* Add ttags

* Add more ttags

* Add etherscan to vote tx

* Add etherscan to airdrop tx

* Remove dead code

* Revise presentation of etherscan link

* Fix pr

dependabot[bot] committed:
Bump eslint-plugin-testing-library from 5.0.5 to 5.0.6 (#546)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ryangoree committed:
add `useAddToWallet` hook (#549)

* add useAddToWallet hook

* Fix typo

ryangoree committed:
`useWorker` refactor (#548)

* Improve useWorker
- Refactor types
- Resolve promises in the worker before posting
- Add error handling
- Add useWorkerPromise

* Add hasRun and remove need for JSON.stringify in useWorker

* Add comment

hnguyen1179 committed:
Add bg to label (#552)

hnguyen1179 committed:
Disable vote choices button if no voting power (#553)

* Disable vote choices button if no voting power

* Refactor code

hnguyen1179 committed:
Group items in proposal details card (#554)

* Copy change: Choice -> Vote

* Group voting power with buttons

hnguyen1179 committed:
Copy change: claimable -> your (#555)

DannyDelott committed:
Add ability to set state after user connects (#556)

dependabot[bot] committed:
Bump postcss from 8.4.7 to 8.4.8 (#551)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump lint-staged from 12.3.4 to 12.3.5 (#550)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ryangoree committed:
ZK refactor (#547)

* Make completePrereqs the default behavior

* Update zk claim intro copy

* Change event names and refactor some logic

* use index for delegate key to avoid errors about repeated keys during testing

* Add useZKProof

* Add wallet connection points

* Replace lib placeholder comments with useZKProof

* Fix component names

* Simplify useZKProof

* Remove `|| undefined` on `Signer`

* Remove scroll ref

* Add onClose and wait for defined account to fire onConnected for ConnectWalletButton

* Add useOnConnected hook

dependabot[bot] committed:
Bump dependency-cruiser from 11.3.1 to 11.4.0 (#557)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ryangoree committed:
Cleanup (#563)

* Remove unused forum page

* Remove old commented code

* Remove div from LinkButton and move attributes to outer tag

* Add type for PopoverButton render fn

sentilesdal commented:
#### 📝 Changed routes:


| [**/forum**](https://elf-council-frontend-staging-kjp8kfkep-element-finance.vercel.app/forum) | [**/gsc-overview**](https://elf-council-frontend-staging-kjp8kfkep-element-finance.vercel.app/gsc-overview) |
|:-:|:-:|
|<a href="https://elf-council-frontend-staging-kjp8kfkep-element-finance.vercel.app/forum"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/2R6u9wk2lKbtuHXxoZCvan0ox2ElRy6oSy8kGD__6jj9hkiphxfKCAzHJSacSe_KvCzKFf_w70X-LAWhDLzeHg.png&w=600" alt="Screenshot of /forum" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/KHU7svQPeUcb4smA4b9Hkq53flgvHmUffvYGXZfku-I208ogr2doZvkywIYT25nDVwiBRXLKqd9enfwjlwjHIw.png">(view full size)</a> |<a href="https://elf-council-frontend-staging-kjp8kfkep-element-finance.vercel.app/gsc-overview"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/ZfW_3pmSeWVgbteOj6dJnZ5XBtsBkjGxHbLZhAHBlMobpqG45vS0xwpZ1Ehdj0veglDjuwhZ4QHc9Fzy_9HrlQ.png&w=600" alt="Screenshot of /gsc-overview" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/bW0KG3ykE3YD-zEoAIQ_yKHcRSIDxwSJQP5tByPwv0Ufw643XWDfd9ma6aVN_kde0Euq1i5FkjMaRqkcSHiwiQ.png">(view full size)</a> |



| [**/gsc-proposals**](https://elf-council-frontend-staging-kjp8kfkep-element-finance.vercel.app/gsc-proposals) | [**/**](https://elf-council-frontend-staging-kjp8kfkep-element-finance.vercel.app/) |
|:-:|:-:|
|<a href="https://elf-council-frontend-staging-kjp8kfkep-element-finance.vercel.app/gsc-proposals"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/6VcleEbnM9QpW6zz_ivO5eWITVANZ5E97ICMLJdWubQOC34DMPgi3dpQDb3ovN_6Z71EZ4ceWMOGKv06De6mLg.png&w=600" alt="Screenshot of /gsc-proposals" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/XogSCJP1Mhqjj64OknYn1qoctfIGd__x5UnPJGOQkUMw-73TYD4E_0LG1JtqVz3tuPhJRkjrfCxvQNi2_f6cFA.png">(view full size)</a> |<a href="https://elf-council-frontend-staging-kjp8kfkep-element-finance.vercel.app/"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/6ATO4DBfqB_gyvKk0rUadKm6_oZqf_v8pHjIgEPP_Z2fLMDnsk1EvgTGuoDsRdfxDJH0BA_Rje9oWBpzsKEUMA.png&w=600" alt="Screenshot of /" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/HTYztHUDDwmlBZS1D3L0HPVebcrd7dRxWPSKOaKFlup78OufKmnBO1OjlfzUlanlsFExm4fg6vFc8AJ6_yixQw.png">(view full size)</a> |



| [**/landing**](https://elf-council-frontend-staging-kjp8kfkep-element-finance.vercel.app/landing) | [**/proposals**](https://elf-council-frontend-staging-kjp8kfkep-element-finance.vercel.app/proposals) |
|:-:|:-:|
|<a href="https://elf-council-frontend-staging-kjp8kfkep-element-finance.vercel.app/landing"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/giMV1lx4PcKDaMtIpgGFMOIx_I7gv6x2t7T7I8Q09NWIAnGUub2pj4GUff8NJfi0pCep7WkyAHdUodEs8GE5QA.png&w=600" alt="Screenshot of /landing" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/3bZLdXRkRiSa-2GCJAJ1YVlkzHIc48PbtWs0dwAsW4NzWmbWk43_ta07K5gCc2kncwCjBcuasR5Drk_OfmWG5g.png">(view full size)</a> |<a href="https://elf-council-frontend-staging-kjp8kfkep-element-finance.vercel.app/proposals"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/amUKEGHJD8BnjqOZTsPwhdDCvMIxfZ-_V6xVe_URMzccA3YWKo1qfqyd57-NiXElK911amCb51JlbrSZMYrXEA.png&w=600" alt="Screenshot of /proposals" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/nE5Hu66KI_UhFIrDXS6RUY9YCFuHXhe6zjSfF-Ki9Zf85ksvqxiOqtpUeJIcCxirNtKIKMYaZheL9ITGhDF5Sg.png">(view full size)</a> |



And 3 other routes:
- [**/vesting-claim**](https://elf-council-frontend-staging-kjp8kfkep-element-finance.vercel.app/vesting-claim)
- [**/zk/discord**](https://elf-council-frontend-staging-kjp8kfkep-element-finance.vercel.app/zk/discord)
- [**/zk/github**](https://elf-council-frontend-staging-kjp8kfkep-element-finance.vercel.app/zk/github)



Commit 5daa8667672ae6871e191f586e50aba6e0f9f25c (https://elf-council-frontend-staging-kjp8kfkep-element-finance.vercel.app).

ryangoree committed:
Single soruce of truth for URLs (#562)

* Create single source of truth for elfi urls

* Add medium url and change "resources" to "docs"

* Convert urls object to enum

ryangoree committed:
Cleanup SVGs (#564)

* Add config for importing SVGs

* Use new SVG components to inline SVGs

* remove double icons (was for testing)

* Use @babel/types and add an accessible title prop to all SVGs

* Touch up comments

* Move SVGs to base

* Fix broken import

* Move home icon in Footer

hnguyen1179 committed:
Touch up proposal details card UI  (#565)

* Disable icon tooltip in proposal details card

* Add disable condition on vote button

* Add scrolling to the proposal description

* Add left align to title

* Adjust font and external link positions

* Add annotation

* More annotation

* Rearrange html structure

* Nits

* Move status label on smaller screens

* Move label on smaller screen down

* Fix spacing

* Add better accessibility

* Fix responsive issue with items clumping

* Cleanup comments

* Css polishing

* Touch up quorum

* Refactor with Tag component

* Css cleanup & expand min-width of proposal details

* Add greater contrast & buttonVariant prop

* Standardized vote button size for all options

* Condensed labels

* Change popover color & ui fixes

* Fix capitalization discrepancy

ryangoree committed:
landing page (wip) (#561)

* Add reward button variant from Figma

* add usePopperWithRefs hook

* Add telegram icon

* Add .vscode to .gitignore

* Add landing page

* Add words to workspace spellcheck settings

ryangoree committed:
Make landing page responsive (#566)

* Make landing page responsive

* Fix mobile

* Prevent menu from closing early on mobile

dependabot[bot] committed:
Bump merkletreejs from 0.2.30 to 0.2.31 (#559)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @testing-library/react from 12.1.3 to 12.1.4 (#560)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ryangoree committed:
Cleanup (#567)

* Fix Card type and move destructuring to the params

* Remove comment from useWorker

* Change useOnConnected.tsx to useOnConnected.ts

hnguyen1179 committed:
Fix proposal details display bug (#568)

* Fix proposal details display bug

* Function rename for clarity

ryangoree committed:
Add address screening (#573)

hnguyen1179 committed:
Change tooltip text (#569)

* Change tooltip text

* Edit definitions and move to single source of truth

* Move tooltip defs out of base

* Use singular to follow conventions

Co-authored-by: Ryan Goree <goree.ryan@gmail.com>

hnguyen1179 committed:
Fix persisting states between proposals (#572)

* Force "Modify vote" button text to stack

* Fix persisting vote selection between proposal details

* Disable loading on button if erroring

* Fix rendering condition for submission button & standardized width

* Fix same etherscan link persisting between proposals

* Fix up documentation

* More styling to submit button & change conditional logic

* Add queryKey

* Remove console.log

* Change conditional logic

* Add more conditional logic

* Add undefined return type to useBallot & cleanup

* Remove unnecessary ?.

* Update pr

* Remove unnecessary undefined checks

dependabot[bot] committed:
Bump ethers from 5.5.4 to 5.6.0 (#570)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump autoprefixer from 10.4.2 to 10.4.4 (#574)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump chromatic from 6.5.1 to 6.5.3 (#571)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

hnguyen1179 committed:
Bug fixes & add modal to proposal page (#577)

* Add modal view to proposal details card

* Fix empty past proposal state not displaying

* Cleanup & fix issue with modal opening on empty proposal lists

* Refactor & export out CloseButton component

* Correct width of proposal details

* Remove unused props

* Remove opinionated css

* Add placeholder for empty proposal detail

* Fix issue with etherscan not displaying on initial load

* Fix pr

* Fix up function returning undefined

hnguyen1179 committed:
Remove profile button (#578)

* Remove profile button

* Cleanup

* Fix up choose button display

ryangoree committed:
ENS Integrations (#579)

* Add `useFormattedWalletAddress` and `useResolvedEnsName`

* Use new hooks

* Remove unused code

* Remove unused line

* Integrate ENS with delegate page input

ryangoree committed:
Add hover states to article links (#580)

hnguyen1179 committed:
Replace council logo w element logo (#582)

hnguyen1179 committed:
Code clean up, ui touch ups and refactor (#584)

* Refactor sidebar

* Add CloseButton component & css fix

* Remove truncate from proposal description

* Adjust spacing for wallet connect buttons

* Add hover state & ExternalLink component

* Adjust spacing

* Revise logic

* Refactor code

* Add spacing

* More refactor

* Make hover full width

* Fix overflow issue

* Remove self closing tag from non-void elements

* Fix issue with outline being cut off

* Fix nits

ryangoree committed:
add higher retry count to useAddressScreening (#586)

* add higher retry count to useAddressScreening

* Edit zk claim date

hnguyen1179 committed:
Add error catching to isEligible (#585)

* Catch error from isEligible

* Fix up PR & add comment

dependabot[bot] committed:
Bump stylelint from 14.5.3 to 14.6.0 (#576)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

DannyDelott committed:
Remove airdrop claim limit (#588)

dependabot[bot] committed:
Bump @blueprintjs/popover2 from 0.13.0 to 1.0.0 (#581)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

DannyDelott committed:
Remove merkle server environment variable (#589)

DannyDelott committed:
Remove unused addresses.json file, prefer from tokenlist repo instead (#592)

sentilesdal committed:
Update elf-council-typechain (#593)

ryangoree committed:
Update zk claim date (#594)

ryangoree committed:
Update landing page links (#595)

ryangoree committed:
Update FAQs (#596)

* Update FAQs

* Add key to link for `jt` and change name

DannyDelott committed:
Add empty mainnet delegates.json (#597)

sentilesdal committed:
add mainnet proposals json (#598)

hnguyen1179 committed:
Update favicon and page titles (#583)

Co-authored-by: Danny Delott <dannydelott@gmail.com>

ryangoree committed:
Copy updates (#599)

ryangoree committed:
remove disclosures (#600)

ryangoree committed:
Ryan cleanup (#601)

* Use singular for enum to follow conventions

* Fix useAddressScreening

* Remove unused icon

ryangoree committed:
Remove header on zk flow (#603)

ryangoree committed:
Fix buttons to match auto-next-step behavior (#602)

ryangoree committed:
Use diff keys for client and server (#606)

* Use diff keys for client and server

* Change key name

hnguyen1179 committed:
Add ElfiverseBanner to Overview Page (#604)

* Add styling to banner

* Change breakpoint from lg -> xl

* Align image right

* Export component into separate file

* Update elfiverse url

* Add count for remaining elf nfts

* Remove extra line

* Expand out FAQ on column view

* Add minting status

* Rename useTokenBalanceOf -> useNFTTokenBalanceOf

* Import clean ups

* Fix gap width

* Fix whitelist bug

DannyDelott committed:
Add mainnet delegates (#607)

ryangoree committed:
update walletconnect (#608)

ryangoree committed:
Hide NFT whitelist countdown after 0 remaining (#610)

* update env sample

* Hide nft banner when countdown reaches 0

hnguyen1179 committed:
Fix wallet bug (#609)

* Fix bug & add todo

* Rename NftTokenContractAddresses -> NFTTokenContractAddresses

* Rename fix

* Rename fix 2

ryangoree committed:
Hide if not relevant (#612)

hnguyen1179 committed:
Update starting block number for whitelist nft (#611)

* Update block number

* Fix pr

* Fix import

ryangoree committed:
Fix nft banner hide logic (#613)

cashd committed:
Vesting Vault Delegation page (#615)

* useDelegationVesting hook

* vesting delegation page

this is just a copy using a different hook.

I am copying to not interfere with the main delegation page for launch

* useDelegateVestingHook

* Update useDelegateVesting.ts

* Update useDelegationVesting.tsx

cashd committed:
Update ElfiverseBanner.tsx (#617)

hnguyen1179 committed:
Update twitter share text (#616)

cashd committed:
Case issue merkle airdrop cashd (#618)

* Update AirdropPage.tsx

* lowercasing

* un lower

* Update AirdropPage.tsx

ryangoree committed:
Update deps (#620)

cashd committed:
elfiverse counter fix (#621)

* Update getRecentDelegators.ts

* Update getRecentDelegators.ts

ryangoree committed:
fix delegate count (#624)

* FIx delegate count filtering

* Fix delegate count

cashd committed:
remove elfiverse banner (#636)

hnguyen1179 committed:
Fix favicon issue  (#641)

* Fix favicon not showing up

* Add back in error catching

* Cleanup favicon with custom _document.tsx

DannyDelott committed:
Revert to http for provider connection (#649)

dependabot[bot] committed:
Bump @fontsource/rubik from 4.5.3 to 4.5.6 (#590)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

hnguyen1179 committed:
Add coinbase connection  (#648)

* Add coinbase connector

* Add close connection functionality

* Fix typo

* Fix import typo

* Clean up

* Fix warnings: Add rule exceptions

* Fix warning: whiteText replaced with textClassName

* Fix warning: remove unused import

* Fix warning: comment out currently unused code

* Fix warnings: Add rule exceptions

* Fix warning: remove unused import

* Undo removal of signer

ryangoree committed:
Update date for first zk round (#655)

ryangoree committed:
Fix nft link (#665)

sentilesdal committed:
Update mainnet snapshot URL (#653)

hnguyen1179 committed:
Update overview total delegates label (#666)

ryangoree committed:
new issue link (#736)

ryangoree committed:
Edit github zk airdrop instructions (#1028)

ryangoree committed:
Integrate zk contract (#650)

* add contract

* Add useClaimableAmount hook

* Add useAlreadyClaimed hook

* Add useClaimAndDelegate hook

* Handle undefined key

* Refactor transaction step to call contract

* Add word to spell check

* wip

* Include all contracts

* Update addresses

ryangoree committed:
Fix key and secret for proof (#1054)

sentilesdal committed:
fix useQueryVotePower, useZKProof (#1055)

ryangoree committed:
Change proof gen time est (#1057)

sentilesdal committed:
subtract foundation and team accounts from circulating vote power calculation (#1059)

sentilesdal committed:
fix voting power calcs (#1060)

* update testnet addresses/tokenlist

* remove unused file

* fix useQueryVotePower

sentilesdal committed:
change label (#1061)

Padraic-O-Mhuiris committed:
Added useFeatureFlag hook (#1062)

sentilesdal committed:
Stubs out the GSC Overview and Proposals pages (#1064)

sentilesdal committed:
GSC Overview - Show GSC members and candidates (#1065)

* update testnet addresses/tokenlist

* list gsc members

* show candidates

* return queryobserverresult

sentilesdal committed:
Add JoinGSCButton to PortfolioCard on the OverviewPage (#1066)

* don't bork if there is no internet connection

* add join button

* wire up join button

* only show if hasGSCFlag

* add progress bar

* disable join is already on GSC

* return envelope

ryangoree committed:
Remove batch 1 ZK date (#1068)
